client/src/components/CharityDetalis/index.jsx

```diff
@@ -50,13 +50,26 @@ class CharityDetalis extends Component {
       mentionOfTheory: '',
     },
     loading: true,
+    found: false,
   };
 
   componentDidMount() {
     const { id } = this.props.match.params;
+    if (!/[0-9]/g.test(id)) {
```

Comment:
@othman-shamla Not a requirement, but I find it helpful to add a comment to lines like these (where there's a regex and it's not immediately obvious what it's looking for), explaining what it does or giving an example of what it might match (and why).